src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -4706,42 +4717,45 @@ select clausesByQuery.First())
             var containingTypeDeclaration = TryGetContainingTypeDeclaration(newMemberBody);
             var isInInterfaceDeclaration = containingTypeDeclaration != null && IsInterfaceDeclaration(containingTypeDeclaration);
 
-            foreach (var newLambda in newMemberBody.DescendantNodesAndSelf())
+            var newHasLambdaBodies = newHasLambdas;
+            while (newHasLambdaBodies)
             {
-                if (TryGetLambdaBodies(newLambda, out var newLambdaBody1, out var newLambdaBody2))
+                var (newLambda, newLambdaBody1, newLambdaBody2) = oldLambdaBodyEnumerator.Current;
```

Comment:
Using the old enumerator here looks wrong, because nothing uses the new enumerators current value.. but the lower loop says "old has _more_ lambdas" which is doing my head in.

```suggestion
                var (newLambda, newLambdaBody1, newLambdaBody2) = newLambdaBodyEnumerator.Current;
```

Reply:
Good catch. Thanks!